### PR TITLE
FIX: Account for additional lines in cisco_nxos_bgp_summary

### PR DESCRIPTION
**Pull request by asik8888**
Few lines in cisco_nxos_show_ip_bgp_summary output error out while parsing. The following changes can help with reading those additional lines seen in "show ip bgp summary" commands.  

Files modified:
cisco_nxos_show_ip_bgp_summary.textfsm
cisco_nxos_show_ip_bgp_summary.raw
cisco_nxos_show_ip_bgp_summary.yml

### CODE REVIEW

**Comment by mjbear** [comment]
Hello @asik8888
Thank you for opening a PR.
The community here was welcoming to me so I intend to pay it forward here.

There are a few items missing from your PR that mean it needs some work before the maintainers will consider merging it. Please allow me to explain.

Please provide a PR description. It is important to provide a brief summary of what your changes accomplish.

There are several bits of documentation in [dev docs](https://github.com/networktocode/ntc-templates/tree/master/docs/dev) and [user docs](https://github.com/networktocode/ntc-templates/tree/master/docs/user)
Specifically:
* [Contributing guidelines](https://github.com/networktocode/ntc-templates/tree/master/docs/dev/contributing.md)
* [Test data is a requirement](https://github.com/networktocode/ntc-templates/tree/master/docs/user/faq.md#i-simply-want-to-add-my-template-to-the-project-i-do-not-want-to-add-all-of-these-tests-and-controls-can-i-just-do-so)
* Details on setting up your [development environment](https://github.com/networktocode/ntc-templates/blob/master/docs/dev/dev_environment.md)
    * [Helper scripts for parsing the raw output](https://github.com/networktocode/ntc-templates/tree/master/docs/dev/dev_parser.md#development-helper-scripts) you provide

I hope this helps guide you.
If anything needs clarified after you've looked at the documentation, please ask. :+1:

File: ntc_templates/templates/cisco_nxos_show_ip_bgp_summary.textfsm
Comment (mjbear) [inline ntc_templates/templates/cisco_nxos_show_ip_bgp_summary.textfsm:20]:
```suggestion
  # ignoring all lines ending in bytes of memory
```

**Review by mjbear — COMMENTED** [review]
Regarding the first suggestion, it's more readable to put a space between the # symbol and the comment.

On the second suggestion, it's best to use regexes to account for white space just in case it was to ever change down the road.

(Flexibility by way of regexes means we hopefully shouldn't find it break often as vendors possibly change their output formatting. It's also a middle ground :balance_scale: between flexibility and making the regexes overly complex.)

**Comment by asik8888** [comment]
Hello Mjbear, 

Thank you for the helpful suggestions, I appreciate your input in helping me. I have resolved the conflicts from the above two suggestions, please let me know if I can do anything further for this PR

**Comment by mjbear** [comment]
> Few lines in "show ip bgp summary" output error out while parsing due to the following lines not being handled correctly in the template. These changes can help with reading those additional lines seen in "show ip bgp summary" commands.

@asik8888 
You might update the title and description a bit to call out which template.

It's tough to really tell without the test data so I'll probably botch my "title" suggestion below until I see the test data. :sweat_smile: 

_Title:_
Account for additional lines in nxos bgp summary

_Body:_
Few lines in **cisco_nxos** "show ip bgp summary" output error out while parsing due to the following lines not being handled correctly in the template. These changes can help with reading those additional lines seen in "show ip bgp summary" commands.

**Comment by asik8888** [comment]
Hello @mjbear,

Added the raw data and yml file to the tests folder. Invoke test, passed.
test

Updated the title and body. Let me know if you would like it in a different format or any other changes. 

Thank you for the help!

**Comment by mjbear** [comment]
> Hello @mjbear,
> 
> Added the raw data and yml file to the tests folder. Invoke test, passed. test
> 
> Updated the title and body. Let me know if you would like it in a different format or any other changes.
> 
> Thank you for the help!

Hello @asik8888,

Apologies for not providing more guidance about the test data.
Adding new test files is oftentimes better than modifying existing ones. (more on that below) :wink:

I would would strongly recommend retaining the existing tests/cisco_nxos/show_ip_bgp_summary/cisco_nxos_show_ip_bgp_summary.raw in its original form.

Then from there add another file with your new test data (example: tests/cisco_nxos/show_ip_bgp_summary/cisco_nxos_show_ip_bgp_summary2.raw).

:bulb: **This way the test data includes not only the old example, but also your new example. This helps provide a broader test base so the project testing has a chance at catching any unexpected errors/regressions.**

Hope that helps!

**Comment by mjbear** [comment]
@asik8888 
I hope you're doing well.

I dropped a PR over on your fork so you should be able to easily merge in those tweaks and update this PR.
Please review https://github.com/asik8888/ntc-templates/pull/1 when you have a chance.
Thank you!

**Comment by asik8888** [comment]
Thank you for input and creating a pull request. sorry for the delay in adding the required files in order. Got busy on something else. I apologize. Have merged the pull request.

**Review by jmcgill298 — APPROVED** [review]

**Comment by asik8888** [comment]
Hi @jmcgill298 

was wondering when this will be released?

**Comment by mjbear** [comment]
> Hi @jmcgill298
> 
> was wondering when this will be released?

Hello @asik8888 
I'll provide some observations here, but I can't speak for the NTC team as to when the next ntc-templates release will be.

The changes you brought forward were merged into the master branch so it is available, but just not yet part of a versioned release. You could `git clone` the repo and point your scripts at that clone if you chose to.

Now as a longer term item a release is certainly more desirable. Seeing as how [the last release (7.3.0) was two weeks ago](https://github.com/networktocode/ntc-templates/releases/tag/v7.3.0) it is possible (though no guarantee) that a release may be coming in a week or two.

**Comment by asik8888** [comment]
Thank you for the update @Michael bear. Appreciate it

**Comment by jmcgill298** [comment]
its out now

**Comment by asik8888** [comment]
Thank you Jacob!

**Pull request merged**
